frontend/src/app/applications/applications.page.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @description
+ *    This page holds the apllications to be finalized and sent to HR
+ */
+import { Component, OnInit } from '@angular/core';
+import { NavController } from '@ionic/angular';
+
+import { FormsData } from '../model/forms-data.model';
+import { StorageHandlerService } from '../services/storage-handler.service';
+
+@Component({
+  selector: 'app-applications',
+  templateUrl: './applications.page.html',
+  styleUrls: ['./applications.page.scss'],
+})
+export class ApplicationsPage implements OnInit {
+  countFinalized = 0;
+  countNotFinalized = 0;
+  notFinishedEntry: FormsData[];
+
+  constructor(
+    private navController: NavController,
+    private storage: StorageHandlerService
+  ) {}
+
+  /**
+   * In this method all the applications stored in the local DB are fetched
+   */
+  ngOnInit(): void {
+    this.storage
+      .getAllItems<FormsData>(this.storage.applicantDetailsDb)
+      .then((data) => {
+        this.fetchApplications(data);
+      });
+  }
+
+  /**
+   * In this method finalized and not finalized application are rendered seperately
+   */
+  fetchApplications(inp: Array<FormsData>): void {
+    this.countFinalized = inp.filter((val) => {
+      return val.isRated === 1;
+    }).length;
+
+    this.notFinishedEntry = inp.filter((x) => x.isRated === 0);
+
+    this.countNotFinalized = this.notFinishedEntry.length;
+  }
+  /**
+   * In this method navigation to HR rating is handled.
+   */
+  goToRating(id: string): void {
+    this.navController.navigateForward(['/rating'], { state: { id } });
```

Comment:
Just out of curiosity: Is there any reason why you put the ID into the state, instead of just using it directly in the URL? The id is, after all, just a string.

Reply:
Ya it can be implemented as dynamic route as well. But I didn't feel the necessity to implement it, as just passing id in state solves the needs. Also In future if we want some other data, we can send it. Otherwise an extra step of querying record is needed.